### PR TITLE
Add HybridDictionary.Keys and HybridDictionary.Values ICollection tests

### PR DESCRIPTION
**Pull request by hughbe**
/cc @ianhays @stephentoub

### CODE REVIEW

File: src/System.Collections.Specialized/tests/HybridDictionary/HybridDictionary.ValuesTests.cs [hunk 1]
```diff
@@ -2,39 +2,44 @@
...
+    public class HybridDictionaryValuesTests : ICollection_NonGeneric_Tests
     {
-        [Theory]
-        [InlineData(0, 0)]
-        [InlineData(0, 1)]
-        [InlineData(5, 0)]
-        [InlineData(5, 0)]
-        [InlineData(50, 0)]
-        [InlineData(50, 1)]
-        public void CopyTo(int count, int index)
```
Comment (ianhays) [inline src/System.Collections.Specialized/tests/HybridDictionary/HybridDictionary.ValuesTests.cs]: Were we *only* testing the CopyTo function for the Values of HybridDictionary? Ouch.

**Comment by ianhays** [comment]
:+1: :shipit:

**Comment by hughbe** [comment]
Thanks for the review @ianhays.

I'm seeing if we can use these helpers in System.Collections.NonGeneric.
I found a bug in Hashtable CopyTo implementations: similar to the `ListDictionary` fix, `Hashtable.CopyTo`, `Hashtable.Keys.CopyTo` and `Hashtable.Values.CopyTo` also throw an `IndexOutOfRangeException` for a non-zero lower bound. Do you suggest I fix this as we did for Specialized, or nerf the tests. This is a low-impact breaking change, as we already know.

Note that this means there is another inconcistency for HybridDictionary: for count 75 we throw IndexOutOfRangeException, but for count 10, we throw ArgumentException

**Comment by ianhays** [comment]
>I found a bug in Hashtable CopyTo implementations: similar to the ListDictionary fix, Hashtable.CopyTo, Hashtable.Keys.CopyTo and Hashtable.Values.CopyTo also throw an IndexOutOfRangeException for a non-zero lower bound. Do you suggest I fix this as we did for Specialized, or nerf the tests. This is a low-impact breaking change, as we already know.

>Note that this means there is another inconcistency for HybridDictionary: for count 75 we throw IndexOutOfRangeException, but for count 10, we throw ArgumentException

This late in the game we'll probably have to leave it as is or spend a *lot* of time forcing it through a compat review since we're at the RTM stage. Because it's a low-impact change on an established type, it's likely not worth the amount of effort it would take. The inconsistency is unfortunate and bothers me more than it probably should, but leaving it in place is more cost effective.

> Do you suggest I fix this as we did for Specialized, or nerf the tests. 

Nerf em!

**Comment by hughbe** [comment]
My thoughts exactly, cheers

**Pull request merged**